src/index.ts

```diff
@@ -71,17 +71,37 @@ export function resolveBundle(bundle: Bundle, resolvers: Resolver[]): Promise<Re
 			// Now that we have a resolver, add the new files needed to the stream
 			resolver.resolve(bundle)
 			.then((additionalItems) => {
-				additionalItems.map((file: FileInfo) => {
+
+				return Promise.map(additionalItems, (file: FileInfo) => {
 					pack.entry({ name: file.name, size: file.size }, file.contents)
+					if (file.name === 'Dockerfile') {
+						return bundle.dockerfileHook(file.contents.toString())
+					} else {
+						return Promise.resolve()
+					}
+				})
+				.then(() => {
+
```

Comment:
I see, seems like something we should define across the board in the TS guide? So we either do or not but consistently across projects and devs